dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/deduplication/DefaultDeduplicationService.java

```diff
@@ -166,8 +170,11 @@ private void merge(DeduplicationMergeParams params)
         original, duplicate, mergeObject.getTrackedEntityAttributes());
     potentialDuplicateStore.moveRelationships(original, duplicate, mergeObject.getRelationships());
     potentialDuplicateStore.moveEnrollments(original, duplicate, mergeObject.getEnrollments());
-
-    potentialDuplicateStore.removeTrackedEntity(duplicate);
+    try {
+      trackerObjectDeletionService.deleteTrackedEntities(List.of(duplicate.getUid()));
+    } catch (NotFoundException e) {
+      throw new RuntimeException("Could not find TrackedEntity: " + duplicate.getUid());
```

Comment:
can this NFE even happen?
I see in the controller we are getting the original and duplicate TEs from the DB, and in case they are not found, we already throw an exception.
https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/deduplication/DeduplicationController.java#L173

Reply:
I don't think it will ever happen as well.
But as I am not sure if we should use the importer here instead, I didn't want to propagate the exception all the way up and just swallowing the exception seemed too much to me